packager/media/test/test_web_server.cc

```diff
@@ -0,0 +1,261 @@
+// Copyright 2023 Google LLC. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#include "packager/media/test/test_web_server.h"
+
+#include <chrono>
+#include <string_view>
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_format.h"
+#include "nlohmann/json.hpp"
+
+// A full replacement for our former use of httpbin.org in tests.  This
+// embedded web server can:
+//
+// 1. Reflect the request method, body, and headers
+// 2. Return a requested status code
+// 3. Delay a response by a requested amount of time
+
+namespace {
+
+// Get a string_view on mongoose's mg_string, which may not be nul-terminated.
+std::string_view view_mg_str(const mg_str& mg_string) {
```

Comment:
nit: use mixed case https://google.github.io/styleguide/cppguide.html#Function_Names

The same below.

Reply:
Thanks!  It's so hard to keep that rule straight in my head when all the libraries we use are in snake_case.